antarest/core/tasks/service.py

```diff
@@ -42,13 +42,23 @@
 
 logger = logging.getLogger(__name__)
 
-TaskUpdateNotifier = t.Callable[[str], None]
-Task = t.Callable[[TaskUpdateNotifier], TaskResult]
-
 DEFAULT_AWAIT_MAX_TIMEOUT = 172800  # 48 hours
 """Default timeout for `await_task` in seconds."""
 
 
+class ITaskNotifier(ABC):
+    @abstractmethod
+    def __call__(self, message: str) -> None:
```

Comment:
as just discussed, better to use a standard method here, otherwise it's confusing to use this object sometimes as a callable, sometimes as an object which has methods